src/statement.d

```diff
@@ -1994,6 +1998,8 @@ public:
         }
         if (increment)
         {
+            if (increment.op == TOK.TOKcomma)
```

Comment:
What about `for (;; a, b, c)` - could you add a test for this?  And also a test that an illegal comma exp used inside a legal comma exp is still detected.

eg `a, (b = (c, d);`